Find a functionally equivalent code:android.os.RecoverySystem.getTrustedCerts:COMMENT
Method Modifier: private     static      
Comment:/**
 * @return the set of certs that can be used to sign an OTA package.
 */

@@ -1,5 +1,5 @@
 {
-    HashSet<Certificate> trusted = new HashSet<Certificate>();
+    HashSet<X509Certificate> trusted = new HashSet<X509Certificate>();
     if (keystore == null) {
         keystore = DEFAULT_KEYSTORE;
     }
@@ -11,17 +11,17 @@
             ZipEntry entry = entries.nextElement();
             InputStream is = zip.getInputStream(entry);
             try {
-                trusted.add(cf.generateCertificate(is));
+                trusted.add((X509Certificate) cf.generateCertificate(is));
             } finally {
                 is.close();
             }

